src/ring_buffer.rs

```diff
@@ -5,20 +5,21 @@ pub struct RingBuffer<T> {
 }
 
 impl<T: Copy + Default> RingBuffer<T> {
+    /// Create a new ring buffer with a given capacity.
     pub fn new(capacity: usize) -> Self {
         RingBuffer {
             buffer: vec![T::default(); capacity],
             head: 0,
             tail: 0,
         }
     }
-
+    /// Reset the ring buffer to its initial state.
     pub fn reset(&mut self) {
         self.buffer.fill(T::default());
         self.head = 0;
         self.tail = 0;
     }
-
+    /// `put` and `peek` write/read without advancing the indices.
     // `put` and `peek` write/read without advancing the indices.
```

Comment:
dupe line of code

Reply:
yeah i copied my original comment to format it into Rust doc but i agree the original comment is probably not necessary anymore